okhttp/src/main/java/okhttp3/internal/ws/RealWebSocket.java

```diff
@@ -396,7 +396,9 @@ synchronized boolean pong(ByteString payload) {
   }
 
   synchronized boolean close(int code, String reason, long cancelAfterCloseMillis) {
-    validateCloseCode(code);
+    if (code != 0) {
+      validateCloseCode(code);
```

Comment:
I assume this behaves differently from making validateCloseCode accept 0 because that would be broken for incoming codes?

Reply:
0 isn't a valid close code, but is acceptable input according to the *public* API `WebSocket.close`, so this seemed correct and safe.  advice welcome...

Reply:
Seems like it's just a documentation error then.

Reply:
I assumed this was an existing decision to mean 0 was something like a socket disconnect, without sending a close reason to the peer.

>    0-999
> 
>       Status codes in the range 0-999 are not used.

Reply:
Seems unlikely. The WebSocket code is at least 3 years old so I would trust the code far more than the docs.

Reply:
option b - https://github.com/square/okhttp/pull/3540